### PR TITLE
[sk] Refactor Column Types

### PR DESCRIPTION
**Pull request by skunichetty**
# Summary
Column type operations (storage of constant values and inference of column types) were refactored to a new `column_types` subpackage to better isolate their logic from the rest of the `data_cleaner` package. 

Additionally, column type constants were replaced with an enum to namespace column type constants, making their use more explicit. Using an enum also makes imports less verbose and easier to manage.

# Tests
Previous unit tests were refactored and ran to confirm that all supported functions of the data cleaner package are not broken.

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_cleaner/column_types/constants.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from enum import Enum
+
+
+class ColumnTypes(str, Enum):
```
Comment (wangxiaoyou1993) [inline mage_ai/data_cleaner/column_types/constants.py:4]: Let's use singular for enum name. `ColumnType`

**Review by wangxiaoyou1993 — APPROVED** [review]

File: mage_ai/data_cleaner/statistics/calculator.py [hunk 2]
```diff
@@ -205,13 +202,25 @@ def __statistics_overview(self, series, col):
...
                 data[f'{col}/min'] = dates.min().isoformat()
+            elif column_type == ColumnType.TEXT:
+                text_series = series_non_null
+                data[f'{col}/avg_str_length'] = text_series.apply(lambda string: len(string)).mean()
+                text_series = text_series.str.split(' ')
+                data[f'{col}/avg_word_count'] = text_series.apply(lambda words: len(words)).mean()
+                text_series = text_series.explode()
+                text_series = text_series[~text_series.str.match('\s*')]
+                data[f'{col}/word_distribution'] = (
+                    text_series.value_counts().head(VALUE_COUNT_LIMIT).todict()
```
Comment (wangxiaoyou1993) [inline mage_ai/data_cleaner/statistics/calculator.py:220]: @skunichetty This line throws error.

Comment (skunichetty) [reply mage_ai/data_cleaner/statistics/calculator.py:220]: This code wasn't even supposed to be in this branch, that's my bad. I can put out a really quick fix for it.

Comment (skunichetty) [reply mage_ai/data_cleaner/statistics/calculator.py:220]: Actually, I have another branch that has this fixed, i'll just pull that branch instead.

Comment (wangxiaoyou1993) [reply mage_ai/data_cleaner/statistics/calculator.py:220]: Let's remove the bad change first.

Comment (skunichetty) [reply mage_ai/data_cleaner/statistics/calculator.py:220]: Fixed at PR here: https://github.com/mage-ai/mage-ai/pull/332

**Pull request merged**